test/e2e/reconciler/partition/partitionset_test.go

```diff
@@ -0,0 +1,332 @@
+		}
+		return false, fmt.Sprintf("expected 1 partition, but got %d", len(partitions.Items))
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected 1 partition")
+
+	t.Logf("Creating a shard part of a third region")
+	shard3 := &corev1alpha1.Shard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "partition-shard-3",
+			Labels: map[string]string{
+				"region": "partition-test-region-3",
+			},
+		},
+		Spec: corev1alpha1.ShardSpec{
+			BaseURL: "https://base.kcp.test.dev",
+		},
+	}
+	shard3, err = shardClient.Cluster(core.RootCluster.Path()).Create(ctx, shard3, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating shard")
+	framework.Eventually(t, func() (bool, string) {
+		partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Get(ctx, partitionSet.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error retrieving partitionSet")
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) && partitionSet.Status.Count == uint16(2) {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 2 partitions, but got %d", partitionSet.Status.Count)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected the partition count to become 2")
+	framework.Eventually(t, func() (bool, string) {
+		partitions, err = partitionClient.Cluster(partitionClusterPath).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error retrieving partitions")
+		if len(partitions.Items) == 2 {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 2 partitions, but got %d", len(partitions.Items))
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected 2 partitions")
+
+	t.Logf("Excluding the shard of the third region")
+	shard3.Labels = map[string]string{
+		"region":   "partition-test-region-3",
+		"excluded": "true",
+	}
+	_, err = shardClient.Cluster(core.RootCluster.Path()).Update(ctx, shard3, metav1.UpdateOptions{})
+	require.NoError(t, err, "error updating shard")
+	framework.Eventually(t, func() (bool, string) {
+		partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Get(ctx, partitionSet.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error retrieving partitionSet")
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) && partitionSet.Status.Count == uint16(1) {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 1 partition, but got %d", partitionSet.Status.Count)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected the partition count to become 1")
+	framework.Eventually(t, func() (bool, string) {
+		partitions, err = partitionClient.Cluster(partitionClusterPath).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error retrieving partitions")
+		if len(partitions.Items) == 1 {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 1 partition, but got %d", len(partitions.Items))
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected 1 partition")
+
+	// The following tests are focused on the admission
```

Comment:
I'd move it to a new separate test that simply checks admission

Reply:
I thought of that but did not do it due to the cost of running a private cluster.